external/CMakeLists.txt

```diff
@@ -140,3 +140,12 @@ if(WITH_BOOTSTRAP)
   endif()
 
 endif()
+
+# libcrypt interface library to optionally use libcrypt if we can find it
+add_library(lokinet-libcrypt INTERFACE)
+pkg_check_modules(CRYPT libcrypt IMPORTED_TARGET)
+if(CRYPT_FOUND AND NOT CMAKE_CROSSCOMPILING)
+  add_definitions(-DHAVE_CRYPT)
+  target_link_libraries(lokinet-libcrypt INTERFACE PkgConfig::CRYPT)
+  message(STATUS "using libcrypt ${CRYPT_VERSION}")
+endif()
```

Comment:
This is currently enable/disabling a feature based on the presence of a dev package on your system.  I think we should feature-gate this behind a cmake option, and if enabled, require libcrypt.  We probably want that disabled by default for any of:
- static build
- Windows
- Android

Also probably worth leaving a "TODO" here to add it to StaticBuild.cmake (i.e. we should do it but defer that for now).